app/models/concerns/orchestration/tftp.rb

```diff
@@ -82,12 +82,12 @@ def default_pxe_render(kind)
   # Adds the host to the forward and reverse TFTP zones
   # +returns+ : Boolean true on success
   def setTFTP(kind)
-    content = generate_pxe_template(kind)
+    content = generate_pxe_template(host.operatingsystem.pxe_loader_map(kind))
     if content
       logger.info "Deploying TFTP #{kind} configuration for #{host.name}"
       each_unique_feasible_tftp_proxy do |proxy|
         mac_addresses_for_provisioning.each do |mac_addr|
-          proxy.set(kind, mac_addr, :pxeconfig => content)
+          proxy.set(kind, mac_addr, {:pxeconfig => content, :targetos => host.operatingsystem.name.downcase})
```

Comment:
What would this do on a Smart Proxy that doesn't support this feature? We have support for using Foreman X.Y with Smart Proxy X.(Y-1) in Satellite (to make upgrading easier for large deployments) so it's always useful to keep that in mind.

Reply:
As we are only adding post parameters here, in case a "old" Smart proxy is used, they are simply ignored.